hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/HazelcastCacheRegionFactory.java

```diff
@@ -48,8 +45,9 @@ public HazelcastCacheRegionFactory(final Properties properties) {
 
     public CollectionRegion buildCollectionRegion(final String regionName, final Properties properties,
                                                   final CacheDataDescription metadata) throws CacheException {
+        /* Collection regions are never versioned, so pass in null for metadata */
         return new HazelcastCollectionRegion<IMapRegionCache>(instance, regionName, properties, metadata,
-                new IMapRegionCache(regionName, instance, properties, metadata));
+                new IMapRegionCache(regionName, instance, properties, null));
```

Comment:
The comment `/* Natural id regions are never versioned, so pass in null for metadata */` above explains why. 

The *only* place `IMapRegionCache` uses the `metadata` for is [here](https://github.com/hazelcast/hazelcast/blob/master/hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/distributed/IMapRegionCache.java#L68) where it gets the entity's `versionComparator`.  The only place this is used for is [here](https://github.com/hazelcast/hazelcast/blob/master/hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/distributed/IMapRegionCache.java#L102) where a version comparison is used to decide which is the latest version of an entity that should be cached. 

For the `HazelcastEntityRegion`, this version comparison makes sense as the latest version of an entity (assuming a version field exists) is always the freshest version of the object.  This is why `buildEntityRegion` passes in the `metadata` to the `IMapRegionCache()` constructor. 

But for `HazelcastCollectionRegion` (which stores the `@OneToMany` etc. relationships between entities and not the entities themselves) and all other region types, this version comparison makes no sense, and it would actually be incorrect to use an entity's version to decide which is the freshest copy of the data.  An entity's relationships to other objects can easily be modified without incrementing its version field, so using the `versionComparator` in non-entity regions could conceivably lead to stale data being cached.